### PR TITLE
Add ARM64 to TestMatrix logic

### LINKED ISSUE #26872: Enable "IsWindowsOnlyTest"s on the Windows.10.Arm64.Open Helix queue
https://github.com/dotnet/aspnetcore/pull/26840 (re)disabled "IsWindowsOnlyTest"s on the Windows.10.Arm64.Open Helix queue because a bunch of work items failed without explicit test failures. At least some of these work item failures were caused by an error in TestMatrix.cs during theory discovery.

```
[xUnit.net 00:00:01.16] IIS.NewShim.FunctionalTests: Exception thrown during theory discovery on 'Microsoft.AspNetCore.Server.IIS.FunctionalTests.StartupTests.HelloWorld'; falling back to single test case.
System.ArgumentException: Arm64
   at Microsoft.AspNetCore.Server.IntegrationTesting.TestMatrix.ResolveDefaultArchitecture() in /_/src/Hosting/Server.IntegrationTesting/src/TestMatrix.cs:line 134
   at Microsoft.AspNetCore.Server.IntegrationTesting.TestMatrix.Build() in /_/src/Hosting/Server.IntegrationTesting/src/TestMatrix.cs:line 101
   at Microsoft.AspNetCore.Server.IntegrationTesting.TestMatrix.GetEnumerator()+MoveNext() in /_/src/Hosting/Server.IntegrationTesting/src/TestMatrix.cs:line 327
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
```

We should properly enable these skipped tests on the Windows.10.Arm64.Open Helix queue. We could do this incrementally by removing the `$(IsWindowsOnlyTest)' != 'true'` check https://github.com/dotnet/aspnetcore/pull/26840/files#diff-fecf6f98b5c2a5cc5c66f045a93d30421b7fedcbdd4cacefc2605cd61a9b89afR46 and then putting `<SkipHelixArm>true</SkipHelixArm>` in the still-failing test projects.

### LINKED ISSUE #26879: [.NET 5 RC2 Regression] HashAlgorithm.ComputeHash is unsupported on 'browser'
### Describe the bug
In .NET 5 RC1 the hash algorithms were not included, cf. https://github.com/dotnet/runtime/issues/42625. This has been fixed (https://github.com/dotnet/runtime/pull/42267): In .NET 5 RC2 we can use hash algorithms again, e.g. to hash the user passwords on the client side. However, the compiler issues a warning, which is a false positive kind of error:

```
Login.razor.cs(140,28): warning CA1416: 'HashAlgorithm.ComputeHash(byte[])' is unsupported on 'browser'
```

I assume that this is an error in the "Browser platform compatibility tooling".

### To Reproduce
Use the following code in Blazor:

```
using SHA512 sha512 = new SHA512Managed();
var data = sha512.ComputeHash(Encoding.UTF8.GetBytes("password"));
```

Then, `dotnet build` will show the warning but the code works fine, though.

### Further technical details
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.2.20479.15
 Commit:    da7dfa8840

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.2.20479.15\

Host (useful for support):
  Version: 5.0.0-rc.2.20475.5
  Commit:  c5a3f49c88

.NET SDKs installed:
  3.1.202 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.2.20479.15 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.2.20475.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.2.20475.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.2.20475.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #26872 by adding ARM architecture checks to TestMatrix.

I'd temporarily enabled the helix arm tests on this PR and didn't see any additional issues.

### CODE REVIEW

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -44,7 +44,7 @@
   <ItemGroup Condition="'$(TargetArchitecture)' == 'arm64' AND '$(IsHelixDaily)' == 'true' AND '$(IsWindowsOnlyTest)' != 'true'">
     <HelixAvailableTargetQueue Include="(Debian.9.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-a12566d-20190807161036" Platform="Linux" />
   </ItemGroup>
-  <ItemGroup Condition="'$(IsWindowsOnlyTest)' != 'true' AND '$(TargetArchitecture)' == 'arm64' AND '$(IsHelixDaily)' == 'true'">
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'arm64'">
```
Comment (Tratcher) [inline eng/targets/Helix.Common.props:47]: @HaoK what was the trick if I wanted to make this queue run on my PR?

Comment (HaoK) [reply eng/targets/Helix.Common.props:47]: Set the pr: none in https://github.com/dotnet/aspnetcore/blob/master/.azure/pipelines/helix-matrix.yml#L2 to include all branches, like https://github.com/dotnet/aspnetcore/blob/master/.azure/pipelines/ci.yml#L27

**Review by HaoK — APPROVED** [review]
Nice!

**Pull request merged**